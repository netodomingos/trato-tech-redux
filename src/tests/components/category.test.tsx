import { getByText, render, screen } from "@testing-library/react"
import Category from "components/category/category"

import assistenteVirtual from 'assets/itens/assistente-virtual.png';
const MockCategory = {
    titulo: 'Teste titulo',
    descricao: 'teste descricao',
    foto: assistenteVirtual,
    favorito: false,
    preco: 1,
    id: 'teste',
    categoria: 'teste categoria'
}

describe('O componente Category', () => {
    test('Quando o componente renderiza', () => {
        const { container } = render( 
            <Category 
                titulo={MockCategory.titulo} 
                descricao={MockCategory.descricao} 
                favorito={MockCategory.favorito} 
                foto={MockCategory.foto}
                preco={MockCategory.preco}
            />
        )

        expect(container).toMatchSnapshot()
    })

    test('Quando o componente renderiza com o titulo', () => {
        render( 
            <Category 
                titulo={MockCategory.titulo} 
                descricao={MockCategory.descricao} 
                favorito={MockCategory.favorito} 
                foto={MockCategory.foto}
                preco={MockCategory.preco}
            />
        )

        const titulo = screen.getByText('Teste titulo')

        expect(titulo).toBeInTheDocument()
    })

    test('Quando o componente renderiza com a descricao', () => {
        render( 
            <Category 
                titulo={MockCategory.titulo} 
                descricao={MockCategory.descricao} 
                favorito={MockCategory.favorito} 
                foto={MockCategory.foto}
                preco={MockCategory.preco}
            />
        )

        const descricao = screen.getByText('teste descricao')

        expect(descricao).toBeInTheDocument()
    })

    test('Quando o componente renderiza a imagem', () => {
        render( 
            <Category 
                titulo={MockCategory.titulo} 
                descricao={MockCategory.descricao} 
                favorito={MockCategory.favorito} 
                foto={MockCategory.foto}
                preco={MockCategory.preco}
            />
        )

        const logo = document.querySelector("img") as HTMLImageElement
        
        expect(logo.src).toContain('assistente-virtual.png')
    })

    test('Quando o componente renderiza a imagem', () => {
        render( 
            <Category 
                titulo={MockCategory.titulo} 
                descricao={MockCategory.descricao} 
                favorito={MockCategory.favorito} 
                foto={MockCategory.foto}
                preco={MockCategory.preco}
            />
        )

        const preco = screen.getByText('R$ 1.00')
        
        expect(preco).toBeInTheDocument()
    })
})