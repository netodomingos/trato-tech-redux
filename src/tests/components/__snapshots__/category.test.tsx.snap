// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`O componente Category Quando o componente renderiza 1`] = `
<div>
  <div
    class="item"
  >
    <div
      class="item-imagem"
    >
      <img
        alt="Teste titulo"
        src="assistente-virtual.png"
      />
    </div>
    <div
      class="item-descricao"
    >
      <div
        class="item-titulo"
      >
        <h2>
          Teste titulo
        </h2>
        <p>
          teste descricao
        </p>
      </div>
      <div
        class="item-info"
      >
        <div
          class="item-preco"
        >
          R$ 
          1.00
        </div>
        <div
          class="item-acoes"
        >
          <svg
            class="item-acao"
            color="#041833"
            fill="currentColor"
            height="24"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(4, 24, 51);"
            viewBox="0 0 1024 1024"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9zM512 814.8S156 586.7 156 385.5C156 283.6 240.3 201 344.3 201c73.1 0 136.5 40.8 167.7 100.4C543.2 241.8 606.6 201 679.7 201c104 0 188.3 82.6 188.3 184.5 0 201.2-356 429.3-356 429.3z"
            />
          </svg>
          <svg
            class="item-acao"
            color="#1875E8"
            fill="currentColor"
            height="24"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(24, 117, 232);"
            viewBox="0 0 576 512"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M504.717 320H211.572l6.545 32h268.418c15.401 0 26.816 14.301 23.403 29.319l-5.517 24.276C523.112 414.668 536 433.828 536 456c0 31.202-25.519 56.444-56.824 55.994-29.823-.429-54.35-24.631-55.155-54.447-.44-16.287 6.085-31.049 16.803-41.548H231.176C241.553 426.165 248 440.326 248 456c0 31.813-26.528 57.431-58.67 55.938-28.54-1.325-51.751-24.385-53.251-52.917-1.158-22.034 10.436-41.455 28.051-51.586L93.883 64H24C10.745 64 0 53.255 0 40V24C0 10.745 10.745 0 24 0h102.529c11.401 0 21.228 8.021 23.513 19.19L159.208 64H551.99c15.401 0 26.816 14.301 23.403 29.319l-47.273 208C525.637 312.246 515.923 320 504.717 320zM408 168h-48v-40c0-8.837-7.163-16-16-16h-16c-8.837 0-16 7.163-16 16v40h-48c-8.837 0-16 7.163-16 16v16c0 8.837 7.163 16 16 16h48v40c0 8.837 7.163 16 16 16h16c8.837 0 16-7.163 16-16v-40h48c8.837 0 16-7.163 16-16v-16c0-8.837-7.163-16-16-16z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
